code/backtracking/gray_code/graycode.cpp

```diff
@@ -0,0 +1,29 @@
+#include<bits/stdc++.h>
+using namespace std;
+vector<int> grayCode(int A) {
+    
+            vector<int> result(1, 0);  
+            for (int i = 0; i < A; i++) {
+                int curSize = result.size();
+                // push back all element in result in reverse order
+                for (int j = curSize - 1; j >= 0; j--) {
+                    result.push_back(result[j] + (1 << i));
+                } 
+            }
+            return result;
+}
+int main()
+{
+	cout<<"Enter the array length: ";
+	int n;

```

Comment:
Read until the user enters a EOF, and don't ask the user for a size if using vectors. (push_back would be useful here).